### PR TITLE
Upgrade guide for 2.15

### PR DESCRIPTION
**Pull request by misohu**
Closes: https://github.com/canonical/mlflow-operator/issues/256

### CODE REVIEW

File: docs/how-to/migrate-v21-v215.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+Migrate Charmed MLflow Version 2.1 to Version 2.15
+==================================================
+
+This guide shows how to migrate Charmed MLflow version 2.1 to version 2.15. This guide assumes you are running the old Charmed MLflow stack version 2.11.
+
+Prerequisites
+-------------
```
Comment (NohaIhab) [inline docs/how-to/migrate-v21-v215.rst:7]: What about upgrading juju and/or microk8s? If it is needed let's link them here

File: docs/how-to/migrate-v21-v215.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+Update MLflow bundle
+--------------------
+
+To update MLflow bundle components from 2.11 to 2.15 please run these commands.
+
+.. code-block:: bash
+
+    juju refresh mlflow-minio --channel=ckf-1.9/stable --trust
+    juju refresh mlflow-mysql --channel=8.0/stable --trust
```
Comment (NohaIhab) [inline docs/how-to/migrate-v21-v215.rst:29]: Isn't it the same version of mysql in the old mlflow bundle?

File: docs/how-to/migrate-v21-v215.rst (hunk 1 above)
Comment (NohaIhab) [reply docs/how-to/migrate-v21-v215.rst:7]: I would say Juju 3.4 and k8s 1.29 is a prerequisite

**Review by NohaIhab — COMMENTED** [review]
thx Michal, a few comments

I think we can also mention upgrading resource dispatcher as an optional step if someone had mlflow integrated with kubeflow, wdyt?

File: docs/how-to/migrate-v21-v215.rst [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+Upgrade MLflow bundle
+--------------------
+
+To Upgrade MLflow bundle components from 2.11 to 2.15 please run these commands:
+
+.. code-block:: bash
+
+    juju refresh mlflow-minio --channel=ckf-1.9/stable --trust
+    juju refresh mlflow-server --channel=2.15/stable --trust
```
Comment (NohaIhab) [inline docs/how-to/migrate-v21-v215.rst:43]:
```suggestion
    juju refresh mlflow-minio --channel=ckf-1.9/stable
    juju refresh mlflow-server --channel=2.15/stable
```
you don't need to grant trust during refresh for charms that already have trust in the old bundle

File: docs/how-to/migrate-v21-v215.rst [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+Upgrade Resource dispatcher (Optional)
+--------------------------------------
+
+If you are running MLflow with Kubeflow you must upgrade your Resource dispatcher deployment. MLflow 2.15 works only with Resource dispatcher version 2.0/stable.
+
+To upgrade your Resource dispatcher charm run these commands:
+
+.. code-block:: bash
+
+    juju refresh resource-dispatcher --channel=2.0/stable --trust
```
Comment (NohaIhab) [inline docs/how-to/migrate-v21-v215.rst:54]:
```suggestion
    juju refresh resource-dispatcher --channel=2.0/stable
```
same as above

**Review by NohaIhab — COMMENTED** [review]
thanks @misohu, almost ready

**Review by NohaIhab — APPROVED** [review]
thx Michal, LGTM

File: docs/how-to/migrate-v21-v215.rst [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+.. tip:: 
+    Before proceeding, you might want to backup MinIO data including your experiments and models. See :ref:`backup` for more details.
+
+Upgrade dependencies
+---------------------
+
+Charmed MLflow 2.15 requires:
+
+1. `MicroK8s`_ version 1.29 or higher.
+2. Juju version 3.4.
```
Comment (afgambin) [inline docs/how-to/migrate-v21-v215.rst:23]: Add link to Juju (product page).

File: docs/how-to/migrate-v21-v215.rst [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+If you do not meet these requirements, please upgrade these components.
```
Comment (afgambin) [inline docs/how-to/migrate-v21-v215.rst:25]: Is here "components" the right word?

File: docs/how-to/migrate-v21-v215.rst [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+.. code-block:: bash
+
+    juju refresh mlflow-minio --channel=ckf-1.9/stable
+    juju refresh mlflow-server --channel=2.15/stable
+
+Upgrade resource dispatcher
+--------------------------------------
+
+Only if you are running MLflow within Kubeflow, you must upgrade your resource dispatcher deployment.
```
Comment (afgambin) [inline docs/how-to/migrate-v21-v215.rst:42]: Add link to the resource dispatcher for more details.

**Pull request merged**